### DISCUSSION TITLE
4 hour interval

### DISCUSSION THREAD

**Discussion by aakusta** [ranaroussi/yfinance#1150] [Development]
Hi, how can we get 4 hour interval data? yahoo finance provides it, yfinance doesnt.

**Comment by ValueRaider** [comment]
Their API doesn't support it, that's what the error message means. If you want it you have to construct from 1H.

**Comment by ValueRaider** [comment]
I had a quick experiment and this code constructs 4h from 1h data:
```
import datetime
import pandas as pd
df = dat.history(..., interval="1h")
df_4h = df.resample('4H', offset=datetime.timedelta(hours=9.5)) # 9.5 hours until market open
df_4h = pd.DataFrame(index=df_4h.last().index, data={
    'Open': df_4h['Open'].first(),
    'High': df_4h['High'].max(),
    'Low': df_4h['Low'].min(),
    'Close': df_4h['Close'].last(),
    'Volume': df_4h['Volume'].sum()
})
```

I'm skeptical of yfinance supporting this because of needing market open time. Fine for most markets and days, but what about unscheduled late opens?

**Reply by aakusta** [reply]
Thanks a lot, I'll try
